taichi/backends/opengl/codegen_opengl.cpp

```diff
@@ -467,17 +480,51 @@ class KernelGen : public IRVisitor {
       emit("{} {} = {}({});", dt_name, stmt->short_name(),
            opengl_data_type_name(stmt->cast_type), stmt->operand->short_name());
     } else if (stmt->op_type == UnaryOpType::cast_bits) {
-      if (stmt->cast_type->is_primitive(PrimitiveTypeID::f32) &&
-          stmt->operand->element_type()->is_primitive(PrimitiveTypeID::i32)) {
+      auto dst_type = stmt->cast_type;
+      auto src_type = stmt->operand->element_type();
+
+      if ((dst_type->is_primitive(PrimitiveTypeID::u32) &&
+           src_type->is_primitive(PrimitiveTypeID::i32)) ||
+          (dst_type->is_primitive(PrimitiveTypeID::u64) &&
+           src_type->is_primitive(PrimitiveTypeID::i64)) ||
+          (dst_type->is_primitive(PrimitiveTypeID::i32) &&
+           src_type->is_primitive(PrimitiveTypeID::u32)) ||
+          (dst_type->is_primitive(PrimitiveTypeID::i64) &&
+           src_type->is_primitive(PrimitiveTypeID::u64))) {
```

Comment:
We can probably leverage `is_integral` and `data_type_size` here to avoid repeated comparisons.